pkg/binder/cache/cache.go

```diff
@@ -17,959 +17,197 @@ limitations under the License.
-		}
-		if currState.pod.Spec.NodeName != pod.Spec.NodeName {
-			// The pod was added to a different node than it was assumed to.
-			klog.InfoS("WARN: Pod was assumed to be on the assumed node, but got added to the current node", "podUID", key, "pod", klog.KObj(pod), "assumedNode", pod.Spec.NodeName, "currentNode", currState.pod.Spec.NodeName)
-		}
-		delete(cache.assumedPods, key)
-		cache.podStates[key].deadline = nil
-		cache.podStates[key].pod = pod
-	case !ok:
-		// Pod was expired. We should add it back.
-		cache.addPod(pod)
-		ps := &podState{
-			pod: pod,
-		}
-		cache.podStates[key] = ps
-	default:
-		return fmt.Errorf("pod %v was already in added state", key)
-	}
-	return nil
-}
-
-func (cache *binderCache) UpdatePod(oldPod, newPod *v1.Pod) error {
-	cache.mu.Lock()
-	defer cache.mu.Unlock()
-
-	podutil.FilteringUpdate(podutil.BoundPod, cache.addPodToCache, cache.updatePodInCache, cache.deletePodFromCache, oldPod, newPod)
-
-	return nil
-}
-
-func (cache *binderCache) updatePodInCache(oldPod, newPod *v1.Pod) error {
-	// A Pod delete event followed by an immediate Pod add event may be merged
-	// into a Pod update event. In this case, we should invalidate the old Pod, and
-	// then add the new Pod.
-	if oldPod.UID != newPod.UID {
-		cache.deletePodFromCache(oldPod)
-		cache.addPodToCache(newPod)
-		return nil
-	}
-
-	key, err := framework.GetPodKey(oldPod)
-	if err != nil {
-		return err
-	}
-
-	currState, ok := cache.podStates[key]
-	switch {
-	// An assumed pod won't have Update/Remove event. It needs to have Add event
-	// before Update event, in which case the state would change from Assumed to Added.
-	case ok && !cache.assumedPods[key]:
-		// Because pod in BinderCache with Spec.NodeName is not empty for binder, so the empty works for scheduler only.
-		// This is tricky, it's better to make implementation of UpdatePod for scheduler and binder separately to make the logic more clear.
-		if currState.pod.Spec.NodeName != "" && currState.pod.Spec.NodeName != newPod.Spec.NodeName {
-			klog.InfoS("Pod updated on a different node than previously added to", "podUID", key)
-			klog.ErrorS(nil, "BinderCache was corrupted and can badly affect scheduling decisions")
-			klog.FlushAndExit(klog.ExitFlushTimeout, 1)
-		}
-		if err := cache.updatePod(oldPod, newPod); err != nil {
-			return err
-		}
-		currState.pod = newPod
-	default:
-		return fmt.Errorf("pod %v is not added to scheduler cache, so cannot be updated", key)
-	}
-	return nil
-}
-
-func (cache *binderCache) DeletePod(pod *v1.Pod) error {
-	cache.mu.Lock()
-	defer cache.mu.Unlock()
+func (cache *binderCache) Dump() *commoncache.Dump {
+	cache.mu.RLock()
+	defer cache.mu.RUnlock()
 
-	if podutil.BoundPod(pod) {
-		cache.deletePodFromCache(pod)
-	}
-	return nil
-}
+	// TODO: Cleanup and extend the Dump.
+	nodeStore, podStore := cache.CommonStoresSwitch.Find(nodestore.Name), cache.CommonStoresSwitch.Find(podstore.Name)
```

Comment:
Try to avoid repeated type assertions
```
nodeStore, podStore := cache.CommonStoresSwitch.Find(nodestore.Name).(*nodestore.NodeStore), cache.CommonStoresSwitch.Find(podstore.Name).(*podstore.PodStore)

nodes := nodeStore.AllNodesClone()
assumedPods := make(map[string]bool, len(podStore.AssumedPods))
for k, v := range podStore.AssumedPods {
		assumedPods[k] = v
	}
```

Reply:
Good catch! Fixed.